src/table/src/metadata.rs

```diff
@@ -162,13 +162,14 @@ impl TableMeta {
     }
 
     pub fn field_column_names(&self) -> impl Iterator<Item = &String> {
-        let columns_schemas = &self.schema.column_schemas();
-        self.value_indices.iter().filter_map(|idx| {
-            let column = &columns_schemas[*idx];
-            if column.is_time_index() {
-                None
+        // `value_indices` is wrong under distributed mode. Use the logic copied from DESC TABLE
+        let columns_schemas = self.schema.column_schemas();
+        let primary_key_indices = &self.primary_key_indices;
+        columns_schemas.iter().enumerate().filter_map(|(i, cs)| {
```

Comment:
Probably I would go with separate filter and map steps here

Reply:
Good point 👍